moveit_ros/warehouse/warehouse/src/import_from_text.cpp

```diff
@@ -33,62 +33,53 @@
  *********************************************************************/
 
 /* Author: Ioan Sucan */
+/* Author: Raphael Bricout */
+/* Author: Shadow Software Team */
 
 #include <moveit/warehouse/planning_scene_storage.h>
 #include <moveit/warehouse/constraints_storage.h>
 #include <moveit/warehouse/state_storage.h>
 #include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+#include <moveit/planning_interface/planning_response.h>
+#include <moveit/planning_interface/planning_interface.h>
 #include <moveit/kinematic_constraints/utils.h>
 #include <moveit/robot_state/conversions.h>
 #include <eigen_conversions/eigen_msg.h>
 #include <boost/program_options/cmdline.hpp>
 #include <boost/program_options/options_description.hpp>
 #include <boost/program_options/parsers.hpp>
 #include <boost/program_options/variables_map.hpp>
+#include <boost/algorithm/string/predicate.hpp>
 #include <ros/ros.h>
 
 static const std::string ROBOT_DESCRIPTION = "robot_description";
 
 void parseStart(std::istream& in, planning_scene_monitor::PlanningSceneMonitor* psm,
-                moveit_warehouse::RobotStateStorage* rs)
+                moveit_warehouse::RobotStateStorage* rs, moveit_msgs::RobotState& startState)
 {
-  int count;
-  in >> count;
-  if (in.good() && !in.eof())
+  std::map<std::string, double> v;
+  std::string joint;
+  std::string marker;
+  double value;
+  in >> joint;
+  while (joint != "." && in.good() && !in.eof())
   {
-    for (int i = 0; i < count; ++i)
-    {
-      std::map<std::string, double> v;
-      std::string name;
-      in >> name;
-      if (in.good() && !in.eof())
-      {
-        std::string joint;
-        std::string marker;
-        double value;
-        in >> joint;
-        while (joint != "." && in.good() && !in.eof())
-        {
-          in >> marker;
-          if (marker != "=")
-            joint = ".";
-          else
-            in >> value;
-          v[joint] = value;
-          if (joint != ".")
-            in >> joint;
-        }
-      }
-      if (!v.empty())
-      {
-        robot_state::RobotState st = psm->getPlanningScene()->getCurrentState();
-        st.setVariablePositions(v);
-        moveit_msgs::RobotState msg;
-        robot_state::robotStateToRobotStateMsg(st, msg);
-        ROS_INFO("Parsed start state '%s'", name.c_str());
-        rs->addRobotState(msg, name);
-      }
-    }
+    in >> marker;
+    if (marker != "=")
+      joint = ".";
+    else
+      in >> value;
+    v[joint] = value;
+    if (joint != ".")
+      in >> joint;
+  }
+  if (!v.empty())
+  {
+    robot_state::RobotState st = psm->getPlanningScene()->getCurrentState();
+    st.setVariablePositions(v);
+    moveit_msgs::RobotState msg;
+    robot_state::robotStateToRobotStateMsg(st, msg);
+    startState = msg;
```

Comment:
You changed a lot of the parsing code in this pull-request.
Is this backward compatible or did you change the format?

Reply:
The current kinetic-devel version only exports the scenes and we haven't change their format. The queries were not exported. The only constraint that it took into account is a link constraint that I don't know how to test. We are adding the option to export and import joint_constraints and pose_constraints.